src/Services/LoggerSubscriber.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace OpenEuropa\EPoetry\Services;
+
+use Phpro\SoapClient\Event\FaultEvent;
+use Phpro\SoapClient\Event\RequestEvent;
+use Phpro\SoapClient\Event\ResponseEvent;
+use Phpro\SoapClient\Events;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Class ePoetryLogPlugin.
+ */
+class LoggerSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @var \OpenEuropa\EPoetry\Services\Logger
+     */
+    protected $logger;
+
+    /**
+     * Constructor.
+     *
+     * @param Logger $logger
+     */
+    public function __construct(Logger $logger)
+    {
+        $this->logger = $logger;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            Events::REQUEST => 'onClientRequest',
+            Events::RESPONSE => 'onClientResponse',
+            Events::FAULT => 'onClientFault',
+        ];
+    }
+
+    /**
+     * @param FaultEvent $event
+     */
+    public function onClientFault(FaultEvent $event): void
+    {
+        $this->logger->logInfo(sprintf(
+            '[epoetry] fault "%s" for request "%s" with params %s',
+            $event->getSoapException()->getMessage(),
+            $event->getRequestEvent()->getMethod(),
+            print_r($event->getRequestEvent()->getRequest(), true)
```

Comment:
Serializing this argument should also be something that the decorator class does, I guess. We could pass those objects on the `$context` parameter here:

```
\Psr\Log\LoggerInterface::info($message, array $context = array());
```